src/google/signer.rs

```diff
@@ -213,6 +217,40 @@ impl Signer {
 
         Err(anyhow!("token not found"))
     }
+
+    fn canonicalize(
+        &self,
+        req: &impl SignableRequest,
+        method: SigningMethod,
+        cred: &Credential,
+    ) -> Result<CanonicalRequest> {
+        let mut creq = CanonicalRequest::new(
+            req,
+            method,
+            None,
+            SigningAlgorithm::Goog4Rsa,
+            "europe-west3".to_string(),
+            "storage".to_string(),
+        )?;
+        creq.build_headers(cred)?;
+        creq.build_query(cred)?;
+
+        debug!("calculated canonical request: {creq}");
+        Ok(creq)
+    }
+
+    pub fn sign_v4(&self, req: &mut impl SignableRequest) -> Result<()> {
+        let credential = self
+            .credential_loader
+            .load_credential()
+            .expect("Did not find credentials for google.");
+        let v4cred: Credential = credential.into();
```

Comment:
I'm not a fan of `From/Into` API, how about create `AwsCredential` here directly? It looks more clear to me.